pytorch_lightning/trainer/trainer.py

```diff
@@ -324,9 +324,11 @@ def __init__(
         self.on_gpu = True if (gpus and torch.cuda.is_available()) else False
 
         # tpu config
-        self.on_tpu = num_tpu_cores is not None
-        self.num_tpu_cores = num_tpu_cores
-        assert num_tpu_cores in [1, 8, None], 'num_tpu_cores can only be 1 or 8'
+        self.on_tpu = tpu_cores is not None
+        self.tpu_cores = tpu_cores
+        assert tpu_cores in [1, 8, None] or len(tpu_cores) == 1, 'tpu_cores can only be 1, 8 or [<1-8>]'
```

Comment:
in docs is also `-1` co is allowed?

Reply:
I haven't come across any functionality that supports `-1` as value to `num_tpu_cores`